sgv2-docsapi/src/main/java/io/stargate/sgv2/docsapi/service/util/configuration/TimeSourceConfiguration.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package io.stargate.sgv2.docsapi.service.util.configuration;
+
+import io.stargate.sgv2.docsapi.service.util.TimeSource;
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Produces;
+
+public class TimeSourceConfiguration {
+
+  @Produces
+  @ApplicationScoped
+  TimeSource system() {
+    return new TimeSource() {
+
+      @Override
+      public long currentTimeMicros() {
+        return currentTimeMillis() * 1000;
```

Comment:
If we are to use this implementation and microseconds, we might consider instead using

    System.nanoTime() / 1000L;

since that would give us accuracy. The one caveat is that the number gotten is not compatible with `System.currentTimeMillis()` as it is not based on epoch (of `1970-01-01` UTC) so either all of these should be based on `nanoTime()` or values should not be mixed.

Reply:
`nanoTime()` can only be used to measure elapsed time, so we can't use it to represent a fixed timestamp.

With that being said, I agree that this deserves some thought; millisecond granularity sounds very coarse, and under high load many concurrent requests could end up using the same timestamp. We could borrow a few ideas from the Java driver's [monotonic generators](https://github.com/datastax/java-driver/blob/4.x/core/src/main/java/com/datastax/oss/driver/internal/core/time/MonotonicTimestampGenerator.java).

I suggest we address this in a follow-up ticket.

Reply:
#1850